### PR TITLE
perf: Load OOO in bulk for getSlots

### PR DESCRIPTION
**Pull request by keithwillcode**
## What does this PR do?

We had this query nested inside of `getUserAvailability` which is run for every user inside of a team event, which can be hundreds. This will instead pre-load this data for all users in 1 query.

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [x] N/A I have updated the developer docs in /docs if this PR makes changes that would require a [documentation change](https://cal.com/docs). If N/A, write N/A here and check the checkbox.
- [x] I confirm automated tests are in place that prove my fix is effective or that my feature works.

## How should this be tested?

- Ensure OOO is still taken into account for users in a team event

### CODE REVIEW

**Review by emrysal — DISMISSED** [review]
😰

**Review by sean-brydon — CHANGES_REQUESTED** [review]
Code works as expected - I think we would need to also add this code in other places that call `getUserAvailability` from what i can see this will stop working in a few places 

V1 API being the main one

**Comment by keithwillcode** [comment]
> Code works as expected - I think we would need to also add this code in other places that call `getUserAvailability` from what i can see this will stop working in a few places
> 
> V1 API being the main one

Oh nice catch @sean-brydon

File: packages/core/getUserAvailability.ts [hunk 1]
```diff
@@ -445,12 +459,74 @@ const _getUserAvailability = async function getUsersWorkingHoursLifeTheUniverseA
...
 
-  const datesOutOfOffice = await getOutOfOfficeDays({
-    userId: user.id,
-    dateFrom,
-    dateTo,
-    availability,
-  });
+  const outOfOfficeDays =
+    initialData?.outOfOfficeDays ??
+    (await prisma.outOfOfficeEntry.findMany({
```
Comment (keithwillcode) [inline packages/core/getUserAvailability.ts:464]: Will be moving this to a repo in separate PR

**Review by sean-brydon — DISMISSED** [review]
Thx for quick fix! LGTM

File: packages/core/getUserAvailability.ts [hunk 2]
```diff
@@ -546,83 +619,11 @@ export interface IOutOfOfficeData {
   };
 }
 
-const getOutOfOfficeDays = async (
-  ...args: Parameters<typeof _getOutOfOfficeDays>
-): Promise<ReturnType<typeof _getOutOfOfficeDays>> => {
-  return monitorCallbackAsync(_getOutOfOfficeDays, ...args);
```
[the remark this answers is not in the document]
Comment (keithwillcode) [reply packages/core/getUserAvailability.ts:552]: Plus I want this function to be pure logic. Will make it easier to move to a domain object in the future.

**Review by sean-brydon — APPROVED** [review]
good spot
